### ISSUE TITLE
When I refactor with webstorm, it doesn't notice the file change. Changing manually doesn't trigger a rebuild.

### ISSUE THREAD

**Issue by hackhat** [whitecolor/ts-node-dev#40]

**Comment by whitecolor** [comment]
Need more info.

**Comment by hackhat** [comment]
What info do you need? Win 10 64bit. Happened again today, I changed another file and it doesn't notice I changed it. I changed it like 10 times and still can't figure out so I need to cold start again.

**Comment by whitecolor** [comment]
I too use on windows, didn't notice any problems with watching try to use `--poll` option.

**Comment by hackhat** [comment]
Got a bit better with `--poll` but I still get the issue. 

So basically I make a change in file a.ts and then the compiler doesn't notice. I keep on changing this file and it never triggers a refresh anymore.

**Comment by whitecolor** [comment]
Is about that big project with a thousand files? Did you try with smaller one?

**Comment by hackhat** [comment]
Yeah it is. I just noticed today again even more impressive.

I add a enum called `A` and use it in the same file. Save the file. Says can't find type `A`. Now I rename the enum to `B` it works! Now rename back to `A`.. and doesn't work. Definitively some kind of cache issue here.

**Comment by hackhat** [comment]
So now every time I add a new property in the interface file this error happens. I think it might have to do with some circular dependency. But the code works after refresh.

**Comment by whitecolor** [comment]
> I add a enum called A and use it in the same file. Save the file. Says can't find type A. Now I rename the enum to B it works! Now rename back to A.. and doesn't work. Definitively some kind of cache issue here.

If you could create repro repo I would look at it.

**Comment by hackhat** [comment]
Here is a basic example

a.ts:

    export interface IState {
      a1: string;
    }
    
b.ts:

    import IState from 'a.ts'

    const state:IState = {
      a1:'a1';
    }
    
Now change the first a.ts to:

    export interface IState {
      a1: string;
      a2: string;
    }
    
And b.ts to:

    import IState from 'a.ts'
    
    const state:IState = {
      a1:'a1',
      a2:'a2',
    }

**Comment by hackhat** [comment]
Yes, `--transpileOnly` I saw this yesterday somewhere and I started using it. Now is flying, really fast and no more this bug.

> Yes I know what is the problem, a.ts contains only interface it is actually not required by ts-node (node) process because it is only used for type checking by typescript compiler, so ts-node-dev doesn't put watch handler on this file.

But why only works sometimes, is not like all the times fails when I make a change, only sometimes.

**Comment by whitecolor** [comment]
> But why only works sometimes, is not like all the times fails when I make a change, only sometimes.

Probably it "works" when you change a source that (required) and uses `a.ts`, so for its new compilation, it will use the new version of `a.ts`, changes of `a.ts` itself is not handled because those files do not have watch handler on them.

> IMO this is a bug, don't you think so?

This is the limitation of the current design, `ts-node-dev` knows about files **required** by node, not by typescript compiler.

**Comment by hackhat** [comment]
I understand it's a limitation but a user can have this bug. Maybe we can try to do something to avoid as it's very annoying and breaking.

Maybe you could refresh all interface only files when compilation fails or something similar.

**Comment by whitecolor** [comment]
I'll think what can be done about it.

**Comment by hackhat** [comment]
Since adding `--transpileOnly ` I didn't had the problem. Just to confirm.

**Comment by robert-king** [comment]
I'm in webstorm on mac. When I edit index.ts, it detects "index.ts has been modified", but on localhost:3000, i still have to hit refresh, it doesn't refresh automatically. 
But!, If I edit lib/views/AnyFile.html, nothing is detected! Is there a simple way to make it detect html changes? 

package.json:
```

{
  "name": "amp-front-end",
  "version": "1.0.0",
  "description": "amp front end for metro",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "tsc": "tsc",
    "build": "node build/index.js",
    "dev": "ts-node-dev --respawn --transpileOnly --poll ./lib/index.ts",
    "prod": "tsc && node ./build/index.js"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@types/express": "^4.16.1",
    "express": "^4.16.4",
    "ts-node-dev": "^1.0.0-pre.32",
    "typescript": "^3.3.3333"
  }
}

```

index.ts:

```
import express = require('express');

// Create a new express application instance
const app: express.Application = express();

var mustacheExpress = require('mustache-express');

// Register '.html' extension with The Mustache Express
app.engine('html', mustacheExpress());

app.set('view engine', 'mustache');
app.set('views', __dirname + '/views'); // you can change '/views' to '/public',
// but I recommend moving your templates to a directory
// with no outside access for security reasons

app.get('/article/:articleId', function (req, res) {
    const data = {
        hello: 'world',
        foo: 'bar'
    };
    res.render('article.html', data);
    //res.send(req.params.articleId);
});

app.listen(3000, function () {
    console.log('Example app listening on port 3000!');
});
```
